xbmc/guilib/StereoscopicsManager.cpp

```diff
@@ -509,6 +509,8 @@ void CStereoscopicsManager::OnPlaybackStarted(void)
   STEREOSCOPIC_PLAYBACK_MODE playbackMode = (STEREOSCOPIC_PLAYBACK_MODE) CSettings::GetInstance().GetInt(CSettings::SETTING_VIDEOPLAYER_STEREOSCOPICPLAYBACKMODE);
   RENDER_STEREO_MODE mode = GetStereoMode();
 
+  g_infoManager.UpdateAVInfo();
```

Comment:
I forgot to describe what it currently does on video playback. When the manager detects that a stereoscopic video is being played (or user toggles from 2D to 3D etc), it tells the GUI to switch into 3D mode as well (if desired by users). GraphicsContext then does it's job and triggers a resolution switch if needed (at least on the PI when using full frame 3D modes)